### ISSUE TITLE
Small instances are not replaced by bigger instances

### ISSUE THREAD

**Issue by runningman84** [aws/karpenter-provider-aws#5315]
### Description

**Observed Behavior**:
Today I have found a cluster with more than 50 nodes (mostly t4g.small). The utilization was quite low like 51% cpu and memory.

These nodes could be easily consolidated to a few big instances. But the consolidation was blocked by
"Can't replace a spot node with a spot node"

In order to cleanup the mess I manually deleted all small nodes and in the now I just have about 15 nodes (It cannot go lower due to topology spreads and az specific stuff which is fine for me). 

I think the problem occurs when a cluster is slowly growing over time which means a new pod just doesn't fit into any existing instances. In this case karpenter just keeps adding a small instances for each new pod.

The huge number of instances create a lot of waste (daemon sets, kubelet reservations, per node licenses, ...) and ebs volumes. This is also linked to
https://github.com/aws/karpenter-provider-aws/issues/5304
https://github.com/aws/karpenter-provider-aws/issues/5157

A workarround would be to specify minimum node sizes but this is also not good because the last node would be always underutilized. 

**Expected Behavior**:
Karpenter should not be blocked because all of our nodes are spot nodes. 

**Reproduction Steps** (Please include YAML):

**Versions**:
- Chart Version: 0.32.2
- Kubernetes Version (`kubectl version`): 1.28.x

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by njtran** [comment]
It seems like this is a duplicate of https://github.com/kubernetes-sigs/karpenter/issues/763 based on the error message. I'll go ahead and close this, but feel free to reopen if you need.
